sources/packages/backend/apps/api/src/route-controllers/report/models/report.dto.ts

```diff
@@ -1,17 +1,25 @@
-import { Allow } from "class-validator";
+import { IsNotEmpty } from "class-validator";
 /**
  * Filter param dynamic json.
  */
 export class ReportFilterParamAPIInDTO {
-  [columnName: string]: unknown;
```

Comment:
This DTO must be using dynamic key name and must not use static property keys. 

The reason is that the reports that ministry view from this page is generated in a dynamic fashion and so is the report filters.

Also this is an exceptional case because of this dynamic behavior that we must do a dry run based validation instead of class validators. 

For more context on these reports please see here https://app.zenhub.com/workspaces/student-information-management-system-5fce9df5aa1b45000e937014/issues/gh/bcgov/sims/559